tests/transaction_receipt_idx_test.go

```diff
@@ -0,0 +1,102 @@
+package tests
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/0xsoniclabs/sonic/gossip/contract/driverauth100"
+	"github.com/0xsoniclabs/sonic/opera"
+	"github.com/0xsoniclabs/sonic/opera/contracts/driverauth"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReceipt_InternalTransactionsDoNotChangeReceiptIndex(t *testing.T) {
+	upgrades := opera.GetSonicUpgrades()
+	net := StartIntegrationTestNetWithJsonGenesis(t, IntegrationTestNetOptions{
+		Upgrades: &upgrades,
+	})
+
+	client, err := net.GetClient()
+	require.NoError(t, err)
+	defer client.Close()
+
+	chainId, err := client.ChainID(t.Context())
+	require.NoError(t, err)
+	gasPrice, err := client.SuggestGasPrice(t.Context())
+	require.NoError(t, err)
+	sender := makeAccountWithBalance(t, net, big.NewInt(1e18))
+
+	startBlockNumber, err := client.BlockNumber(t.Context())
+	require.NoError(t, err)
+
+	numSimpleTxs := 10
```

Comment:
```suggestion
	numSimpleTxs := uint64(10)
```
as everywhere it is used it is casted to `uint64` anyways.

Reply:
It is only used as an `uint64` once in the current version, I prefer to keep it as an `int` and cast it once.